resolve-hostname.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# Starting new instance in VPC often results that `hostname` returns something like 'ip-10-1-1-24', which is
+# not resolvable. Which leads to problems like SparkUI failing to bind itself on start up to that hostname as
+# described in https://issues.apache.org/jira/browse/SPARK-5246.
+# This script maps private ip to such hostname via '/etc/hosts'.
+#
+
+# Are we in VPC?
+MAC=`wget -q -O - http://169.254.169.254/latest/meta-data/mac`
+VCP_ID=`wget -q -O - http://169.254.169.254/latest/meta-data/network/interfaces/macs/${MAC}/vpc-id`
+if [ -n "${VCP_ID}" ]; then
```

Comment:
@voukka Shouldn't this be `if [ -z "${VPC_ID}"]` ? `-n` is true if `$VPC_ID` is not empty and we want to exit if the string is empty ?